src/EventStore.Core/Services/SubscriptionTracker.cs

```diff
@@ -0,0 +1,45 @@
+#nullable enable
+using System;
+using EventStore.Core.Bus;
+using EventStore.Core.Data;
+using EventStore.Core.Messages;
+
+namespace EventStore.Core.Services;
+
+public interface ISubscriptionTracker {
+	void AddSubscription(Guid subscriptionId, string? streamName, long endOfStream);
+	void RemoveSubscription(Guid subscriptionId);
+	void RecordEvent(EventRecord @event);
+	void ProcessEvent(ResolvedEvent @event);
+}
+
+public class SubscriptionMetricHandler(ISubscriptionTracker tracker) : IHandle<StorageMessage.EventCommitted> {
+	public void Handle(StorageMessage.EventCommitted message) => tracker.RecordEvent(message.Event);
+}
```

Comment:
i think the tracker could implement IHandle directly, no need for the extra wrapper